repair/row_level.cc

```diff
@@ -2300,35 +2300,42 @@ static future<> repair_get_full_row_hashes_with_rpc_stream_handler(
         uint32_t repair_meta_id,
         rpc::sink<repair_hash_with_cmd> sink,
         rpc::source<repair_stream_cmd> source) {
-    return repeat([&repair, from, src_cpu_id, dst_cpu_id, repair_meta_id, sink, source] () mutable {
-        return do_with(false, [&repair, from, src_cpu_id, dst_cpu_id, repair_meta_id, sink, source] (bool& error) mutable {
-            return source().then([&repair, from, src_cpu_id, dst_cpu_id, repair_meta_id, sink, source, &error] (std::optional<std::tuple<repair_stream_cmd>> status_opt) mutable {
-                if (status_opt) {
+    std::exception_ptr outer_exception;
+    try {
+        while (std::optional<std::tuple<repair_stream_cmd>> status_opt = co_await source()) {
+            bool error = false;
+            std::exception_ptr ep;
+            try {
                     if (error) {
-                        return make_ready_future<stop_iteration>(stop_iteration::no);
+                        continue;
                     }
-                    return repair_get_full_row_hashes_with_rpc_stream_process_op(repair, from,
+                    auto stop = co_await repair_get_full_row_hashes_with_rpc_stream_process_op(repair, from,
                             src_cpu_id,
                             dst_cpu_id,
                             repair_meta_id,
                             sink,
                             source,
                             error,
-                            std::move(status_opt)).handle_exception([from, repair_meta_id, sink, &error] (std::exception_ptr ep) mutable {
+                            std::move(status_opt));
+                    if (stop) {
```

Comment:
We do not need repair_get_full_row_hashes_with_rpc_stream_process_op to return stop after the refactor. It always returns stop_iteration::no.

Reply:
I believe it was the case before, no? The coroutinization wasn't supposed to change how any of the functions behave.

(in some cases where there are exceptions in something::close(), it could return a different exception, but something::close() shouldn't fail).